manager/apis/app/v1alpha1/fybrikapplication_types.go

```diff
@@ -155,25 +155,34 @@ type DatasetDetails struct {
 	Details serde.Arbitrary `json:"details,omitempty"`
 }
 
-// FybrikApplicationStatus defines the observed state of FybrikApplication.
-type FybrikApplicationStatus struct {
+// AssetState defines the observed state of an asset
+type AssetState struct {
+	// Conditions indicate the asset state (Ready, Deny, Error)
+	// +optional
+	Conditions []Condition `json:"conditions,omitempty"`
 
-	// Ready is true if a blueprint has been successfully orchestrated
-	Ready bool `json:"ready,omitempty"`
+	// CatalogedAsset provides a new asset identifier after being registered in the enterprise catalog
+	// +optional
+	CatalogedAsset string `json:"catalogedAsset,omitempty"`
 
-	// Conditions represent the possible error and failure conditions
+	// Endpoint provides the endpoint spec from which the asset will be served to the application
 	// +optional
-	Conditions []Condition `json:"conditions,omitempty"`
+	Endpoint EndpointSpec `json:"endpoint,omitempty"`
+}
```

Comment:
Shouldn't at least one of the AssetState fields will be required?

Reply:
All status fields are optional.